lib/scarpe/glibui/app.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+class Scarpe
+  # Scarpe::GlimmerLibUIApp must only be used from the main thread, due to GTK+ limitations.
+  class GlimmerLibUIApp < GlimmerLibUIWidget
+    attr_reader :debug
+    attr_reader :control_interface
+
+    attr_writer :shoes_linkable_id
+
+    def initialize(properties)
+      # Is this a thing? Do we care about this?
+      # opts = @control_interface.app_opts_get_override(opts)
+
+      super
+
+      # It's possible to provide a Ruby script by setting
+      # SCARPE_TEST_CONTROL to its file path. This can
+      # allow pre-setting test options or otherwise
+      # performing additional actions not written into
+      # the Shoes app itself.
+      #
+      # The control interface is what lets these files see
+      # events, specify overrides and so on.
+      @control_interface = ControlInterface.new
+      if ENV["SCARPE_TEST_CONTROL"]
+        @control_interface.instance_eval File.read(ENV["SCARPE_TEST_CONTROL"])
```

Comment:
You can decide whether the control interface is a thing you want here. It adds a fair bit of complexity, though it *does* make evented testing much easier.